src/lib/optimizeSvg.ts

```diff
@@ -0,0 +1,24 @@
+import SVGO from 'svgo'
+import fs from 'fs'
+
+let svgOptimizer: SVGO
+
+const fsp = fs.promises
+
+export default async function optimizeSvg(
```

Comment:
Splitting this out will cause a bug if the user is doing multiple builds in a single process (using Vite's programmatic API) and the `svgOpts` are different between builds. For this reason, we need to keep `optimizeSvg` in the scope of the plugin factory.